eventuals/expected.h

```diff
@@ -26,54 +28,96 @@ Overloaded(Fs...) -> Overloaded<Fs...>;
 
 ////////////////////////////////////////////////////////////////////////
 
-// Tag type to identify the result of 'Unexpected(...)'.
+// Helper type to represent an 'Unexpected(...)'.
+template <typename Error_>
 struct _Unexpected final {
-  _Unexpected() = delete;
+  _Unexpected(Error_ error)
+    : error(std::move(error)) {}
+
+  Error_ error;
 };
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename Value_>
-class _Expected final {
+template <typename Value_, typename Errors_ = std::tuple<std::exception>>
+class _Expected {
  public:
-  // Providing 'ValueFrom' in order to compose with other eventuals.
+  static_assert(
+      !tuple_types_contains_v<Value_, Errors_>,
+      "'Expected::Of' value and raised errors must be disjoint");
+
+  // Providing 'ValueFrom' and 'ErrorsFrom' to compose with eventuals.
   template <typename Arg>
   using ValueFrom = Value_;
 
-  // NOTE: providing a default constructor so this type can be used as
-  // a type parameter to 'std::promise' which fails on MSVC (see
-  // https://stackoverflow.com/questions/28991694 which also ran into
-  // this issue; very surprised this hasn't been fixed, apparently
-  // there aren't many users of 'std::promise' on Windows).
-  _Expected() {}
+  template <typename Arg, typename Errors>
```

Comment:
we can make
`template <typename Arg, typename... Errors>...` - that isn't require passing `std::tuple<>` in tests, but need to add a comment that it expects a `tuple` instead of a set of types.

Reply:
I'm not following, can you elaborate?

Reply:
Let's do this in a separate PR!